app/src/main/java/org/greenstand/android/TreeTracker/activities/MainActivity.kt

```diff
@@ -139,6 +146,11 @@ class MainActivity : AppCompatActivity(), ActivityCompat.OnRequestPermissionsRes
         }
     }
 
+    public override fun onDestroy() {
```

Comment:
This is added to ensure `stepCounter.viewLifecycleObserver` is removed when the app is exited unexpectedly without completing the tree capture.